<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtBinFile</class>
 <widget class="QMainWindow" name="QtBinFile">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>517</width>
    <height>659</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>QtBinFile</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupIn">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="hlayoutBin">
         <item>
          <widget class="QPushButton" name="btnBin">
           <property name="text">
            <string>测试用文件</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="txtBin"/>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="hlayoutRadio">
         <item>
          <widget class="QGroupBox" name="groupEndian">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>字节序</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QRadioButton" name="radioLittle">
              <property name="text">
               <string>LittleEndian</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioBig">
              <property name="text">
               <string>BIgEndian</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupPrec">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>浮点数精度</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QRadioButton" name="radioSingle">
              <property name="text">
               <string>SinglePrecision</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioDouble">
              <property name="text">
               <string>DoublePrecision</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupOut">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>预编码数据序列化方式读写测试</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="7" column="4" rowspan="2" colspan="2">
        <widget class="QDoubleSpinBox" name="dbspinFloat16">
         <property name="maximum">
          <double>999.899999999999977</double>
         </property>
         <property name="value">
          <double>689.889999999999986</double>
         </property>
        </widget>
       </item>
       <item row="11" column="6" rowspan="2">
        <widget class="QPushButton" name="btnDoubleW">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="2" column="6">
        <widget class="QPushButton" name="btnQint16W">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="4" column="4" colspan="2">
        <widget class="QSpinBox" name="spinQint32">
         <property name="minimum">
          <number>-99999</number>
         </property>
         <property name="maximum">
          <number>99999</number>
         </property>
         <property name="value">
          <number>-8888</number>
         </property>
        </widget>
       </item>
       <item row="4" column="8">
        <widget class="QLineEdit" name="txtQint32"/>
       </item>
       <item row="13" column="8" rowspan="2">
        <widget class="QLineEdit" name="txtCharR"/>
       </item>
       <item row="15" column="7">
        <widget class="QPushButton" name="btnQStringR">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="7" column="6" rowspan="2">
        <widget class="QPushButton" name="btnFloat16W">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="4">
        <widget class="QLabel" name="labBool">
         <property name="text">
          <string>bool</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="9" column="0" rowspan="2" colspan="4">
        <widget class="QLabel" name="labFloat">
         <property name="text">
          <string>float</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="16" column="7" rowspan="2">
        <widget class="QPushButton" name="btnFontR">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="11" column="4" rowspan="2" colspan="2">
        <widget class="QDoubleSpinBox" name="dbspinDouble">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-9999.000000000000000</double>
         </property>
         <property name="maximum">
          <double>9999.000000000000000</double>
         </property>
         <property name="value">
          <double>-1234.567800000000034</double>
         </property>
        </widget>
       </item>
       <item row="0" column="8">
        <widget class="QLineEdit" name="txtQint8"/>
       </item>
       <item row="3" column="7">
        <widget class="QPushButton" name="btnQuint16R">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="13" column="0" rowspan="2" colspan="4">
        <widget class="QLabel" name="labChar">
         <property name="text">
          <string>char* string</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="7" column="2" rowspan="2" colspan="2">
        <widget class="QLabel" name="labFloat16">
         <property name="text">
          <string>qfloat16</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="6">
        <widget class="QPushButton" name="btnQint8W">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="6" column="8">
        <widget class="QCheckBox" name="chkRbool">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
       <item row="15" column="8">
        <widget class="QLineEdit" name="txtQStringR"/>
       </item>
       <item row="5" column="8">
        <widget class="QLineEdit" name="txtQint64"/>
       </item>
       <item row="2" column="7">
        <widget class="QPushButton" name="btnQint16R">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="11" column="0" rowspan="2" colspan="4">
        <widget class="QLabel" name="labDouble">
         <property name="text">
          <string>double</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="4">
        <widget class="QLabel" name="labInt32">
         <property name="text">
          <string>qint32</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="6">
        <widget class="QPushButton" name="btnQuint8W">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4" colspan="2">
        <widget class="QSpinBox" name="spinQint8">
         <property name="minimum">
          <number>-128</number>
         </property>
         <property name="maximum">
          <number>127</number>
         </property>
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item row="16" column="8" rowspan="2">
        <widget class="QLineEdit" name="txtFont">
         <property name="text">
          <string>字体</string>
         </property>
        </widget>
       </item>
       <item row="9" column="8" rowspan="2">
        <widget class="QLineEdit" name="txtFloat"/>
       </item>
       <item row="11" column="7" rowspan="2">
        <widget class="QPushButton" name="btnDoubleR">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="13" column="7" rowspan="2">
        <widget class="QPushButton" name="btnCharR">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="18" column="6">
        <widget class="QPushButton" name="btnColorW">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="13" column="4" rowspan="2" colspan="2">
        <widget class="QLineEdit" name="txtChar">
         <property name="text">
          <string>Hello char</string>
         </property>
        </widget>
       </item>
       <item row="1" column="7">
        <widget class="QPushButton" name="btnQuint8R">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="3" column="4" colspan="2">
        <widget class="QSpinBox" name="spinQuint16">
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>65520</number>
         </property>
        </widget>
       </item>
       <item row="16" column="6" rowspan="2">
        <widget class="QPushButton" name="btnFontW">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="18" column="8">
        <widget class="QLineEdit" name="txtColor">
         <property name="text">
          <string>颜色</string>
         </property>
        </widget>
       </item>
       <item row="18" column="7">
        <widget class="QPushButton" name="btnColorR">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="4" column="7">
        <widget class="QPushButton" name="btnQint32R">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="9" column="7" rowspan="2">
        <widget class="QPushButton" name="btnFloatR">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="15" column="0" colspan="4">
        <widget class="QLabel" name="labString">
         <property name="text">
          <string>QString</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="16" column="0" rowspan="2" colspan="4">
        <widget class="QLabel" name="labFont">
         <property name="text">
          <string>QFont</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="3" column="6">
        <widget class="QPushButton" name="btnQuint16W">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="4">
        <widget class="QLabel" name="labInt64">
         <property name="text">
          <string>qint64</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="6" column="7">
        <widget class="QPushButton" name="btnBoolR">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="3" column="8">
        <widget class="QLineEdit" name="txtQuint16"/>
       </item>
       <item row="5" column="4" colspan="2">
        <widget class="QSpinBox" name="spinQint64">
         <property name="minimum">
          <number>-99999999</number>
         </property>
         <property name="maximum">
          <number>99999999</number>
         </property>
         <property name="value">
          <number>12345678</number>
         </property>
        </widget>
       </item>
       <item row="5" column="6">
        <widget class="QPushButton" name="btnQint64W">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="6" column="4" colspan="2">
        <widget class="QCheckBox" name="chkBool">
         <property name="text">
          <string>布尔值</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="4">
        <widget class="QLabel" name="labUint16">
         <property name="text">
          <string>quint16(0 ~ 65535)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="8">
        <widget class="QLineEdit" name="txtQint16"/>
       </item>
       <item row="5" column="7">
        <widget class="QPushButton" name="btnQint64R">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="4">
        <widget class="QLabel" name="labInt16">
         <property name="text">
          <string>qint16(-32768 ~ 32767)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="4">
        <widget class="QLabel" name="labQuint8">
         <property name="text">
          <string>quint8(0 ~ 255)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="4" colspan="2">
        <widget class="QSpinBox" name="spinQuint8">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>235</number>
         </property>
        </widget>
       </item>
       <item row="15" column="4" colspan="2">
        <widget class="QLineEdit" name="txtQString">
         <property name="text">
          <string>Hello QString</string>
         </property>
        </widget>
       </item>
       <item row="6" column="6">
        <widget class="QPushButton" name="btnBoolW">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="7" column="8" rowspan="2">
        <widget class="QLineEdit" name="txtFloat16"/>
       </item>
       <item row="18" column="4" colspan="2">
        <widget class="QPushButton" name="btnColor">
         <property name="text">
          <string>选择颜色</string>
         </property>
        </widget>
       </item>
       <item row="7" column="7" rowspan="2">
        <widget class="QPushButton" name="btnFloat16R">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="16" column="4" rowspan="2" colspan="2">
        <widget class="QPushButton" name="btnFont">
         <property name="text">
          <string>选择字体</string>
         </property>
        </widget>
       </item>
       <item row="1" column="8">
        <widget class="QLineEdit" name="txtQuint8"/>
       </item>
       <item row="0" column="7">
        <widget class="QPushButton" name="btnQint8R">
         <property name="text">
          <string>读取</string>
         </property>
        </widget>
       </item>
       <item row="4" column="6">
        <widget class="QPushButton" name="btnQint32W">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="2" column="4" colspan="2">
        <widget class="QSpinBox" name="spinQint16">
         <property name="minimum">
          <number>-32768</number>
         </property>
         <property name="maximum">
          <number>32767</number>
         </property>
         <property name="value">
          <number>-1522</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="4">
        <widget class="QLabel" name="labQint8">
         <property name="text">
          <string>qint8(-128 ~ 127)</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="9" column="4" rowspan="2" colspan="2">
        <widget class="QDoubleSpinBox" name="dbspinFloat">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>9999.999900000000707</double>
         </property>
         <property name="value">
          <double>1234.567800000000034</double>
         </property>
        </widget>
       </item>
       <item row="9" column="6" rowspan="2">
        <widget class="QPushButton" name="btnFloatW">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="15" column="6">
        <widget class="QPushButton" name="btnQStringW">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="13" column="6" rowspan="2">
        <widget class="QPushButton" name="btnCharW">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item row="18" column="0" colspan="4">
        <widget class="QLabel" name="labColor">
         <property name="text">
          <string>QColor</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="11" column="8" rowspan="2">
        <widget class="QLineEdit" name="txtDouble"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextBesideIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionClearAll"/>
   <addaction name="actionWrite"/>
   <addaction name="actionRead"/>
   <addaction name="separator"/>
   <addaction name="actionQuit"/>
  </widget>
  <action name="actionClearAll">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon_rw/qingkongziyuan.svg</normaloff>:/icon_rw/qingkongziyuan.svg</iconset>
   </property>
   <property name="text">
    <string>全部清空</string>
   </property>
  </action>
  <action name="actionWrite">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon_rw/xierushujuku.svg</normaloff>:/icon_rw/xierushujuku.svg</iconset>
   </property>
   <property name="text">
    <string>连续写入文件</string>
   </property>
  </action>
  <action name="actionRead">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon_rw/duqushujuku.svg</normaloff>:/icon_rw/duqushujuku.svg</iconset>
   </property>
   <property name="text">
    <string>连续读取文件</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/icon_rw/tuichu.svg</normaloff>:/icon_rw/tuichu.svg</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>QtBinFile</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>330</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
